capa/features/extractors/drakvuf/call.py

```diff
@@ -0,0 +1,56 @@
+# Copyright (C) 2023 Mandiant, Inc. All Rights Reserved.
+# Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at: [package root]/LICENSE.txt
+# Unless required by applicable law or agreed to in writing, software distributed under the License
+#  is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and limitations under the License.
+
+import logging
+from typing import Tuple, Iterator
+
+from capa.features.insn import API, Number
+from capa.features.common import String, Feature
+from capa.features.address import Address
+from capa.features.extractors.base_extractor import CallHandle, ThreadHandle, ProcessHandle
+from capa.features.extractors.drakvuf.models import Call
+
+logger = logging.getLogger(__name__)
+
+
+def extract_call_features(ph: ProcessHandle, th: ThreadHandle, ch: CallHandle) -> Iterator[Tuple[Feature, Address]]:
+    """
+    this method extracts the given call's features (such as API name and arguments),
+    and returns them as API, Number, and String features.
+
+    args:
+      ph: process handle (for defining the extraction scope)
+      th: thread handle (for defining the extraction scope)
+      ch: call handle (for defining the extraction scope)
+
+    yields:
+      Feature, address; where Feature is either: API, Number, or String.
+    """
+    call: Call = ch.inner
+
+    # list similar to disassembly: arguments right-to-left, call
+    for arg_value in call.arguments.values():
+        if arg_value.isdecimal():
+            yield Number(int(arg_value)), ch.address
+        else:
```

Comment:
are you sure that this can be just a string and a number?

You could also use insistence instead of a try expect block. Let's say that there is an error, then the block will catch it and use String(...), i.e., the error message can be cryptic.

Reply:
Hmm, I couldn't find an efficient way to check whether the value is a hexadecimal. Just catching the possible ValueError and defaulting to returning it as string seemed efficient.

Also yes I think the values should be either a number or a string. If not then Pydantic would detect it early on. It might be good to catch a ValueError specifically however yeah.